### PR TITLE
FH-3607 - Add command `fhc services` to V3 standard and add unit test

### PR DESCRIPTION
**Pull request by camilamacedo86**
JIRA: https://issues.jboss.org/browse/FH-3607

Following the changes.

* Add the V3 standard
* Divide as category 
* Improve output create, update and delete
* Add unit tests

Following the local tests. 

* Command: ````  ./bin/fhc.js services`````

screen shot 2017-05-29 at 11 06 07 pm

* Command: ````  ./bin/fhc.js services list`````

screen shot 2017-05-29 at 11 03 48 pm

* Command: ````  ./bin/fhc.js services read --service`````

screen shot 2017-05-29 at 11 07 02 pm

* Command: ````fhc services update --service=<service> --propName=<propName> --propValue=<propValue>`````

```` 
Camilas-MacBook-Pro:fh-fhc cmacedo$  ./bin/fhc.js services update oqaijkswo4ecdsxzrbur2cbe PropName PropValue
(node) sys is deprecated. Use util instead.
Service 'Test' update successfully with the propName 'PropName' and propValue 'PropValue'
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
```` 
* Command: ```` fhc services data-sources --service=<service> ```

screen shot 2017-05-29 at 11 09 49 pm

Observation: 

The oldest version the same result:

screen shot 2017-05-29 at 11 10 43 pm

JIRA to check the REST service: https://issues.jboss.org/browse/RHMAP-15948
I could not find a service that returned a result. 

* Command: ```` fhc services delete --service=<service> ```

`````
Camilas-MacBook-Pro:fh-fhc cmacedo$  ./bin/fhc.js services delete --service=oqaijkswo4ecdsxzrbur2cbe
(node) sys is deprecated. Use util instead.
Service 'Test' delete successfully
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
`````

* Command: ```` fhc services create ```

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js services create --title=NewTest
(node) sys is deprecated. Use util instead.
Service 'NewTest' create successfully
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js services create --title=NewTest --template=new-service
(node) sys is deprecated. Use util instead.
Service 'NewTest' create successfully
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js services create --title=NewTest --template=new-service --env=dev
(node) sys is deprecated. Use util instead.
Service 'NewTest' create successfully
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
```

Checking into the studio.

screen shot 2017-05-29 at 11 52 03 pm

### CODE REVIEW

File: lib/cmd/fh3/services/data-sources.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    fhc.services.read({service:params.service}, function(err, service) {
+      if (err) {
+        return cb(err);
+      }
+
+      common.doGetApiCall(fhreq.getFeedHenryUrl(), "api/v2/services/" + service.apps[0].guid + "/data_sources", "Error reading Service data sources. ", function(err, dataSources) {
+        if (err) {
+          return cb(err);
+        }
+        params._table = common.createTableForDataSources(dataSources || [], service);
```
Comment (wei-lee) [inline lib/cmd/fh3/services/data-sources.js]: this behaviour is different from the previous implementation I think. Should it only print out in the table format when the `--table` flag is passed?

File: lib/cmd/fh3/services/list.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  },
+  'describe' : {
+    'json' : i18n._('Output into json format')
+  },
+  'customCmd' : function(params,cb) {
+    common.listServices(function(err, services) {
+      if (err) {
+        return cb(err);
+      }
+      params._table = common.createTableForProjects(services);
```
Comment (wei-lee) [inline lib/cmd/fh3/services/list.js]: the same as above

**Comment by wei-lee** [comment]
LGTM. Just a question around the format of the list output.

File: lib/cmd/fh3/services/data-sources.js (hunk 1 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/services/data-sources.js]: Hi @wei-lee thanks for the review. If you check the first comment you will see that the behavior is the same. For default before this changes, the result is the table ( I put a test with the changes and whiteout the changes). Also, the code is doing the same, which was only refactored in this case.

File: lib/cmd/fh3/services/list.js (hunk 2 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/services/list.js]: It is the same of the above too. The default behavior is to return the table before the changes. Following the local test without any change. 

screen shot 2017-05-30 at 8 55 03 am

**Pull request merged**